### PR TITLE
HLS Module

### PR DESCRIPTION
**Pull request by sverekar**
Hi @evan-hughes , @ananner @bmatthew 

This PR is for EW compatible external HLS module. 

For this release, we have targeted manifest-personalization, bumper insertion and live program replacement usecases.
Kindly review the changes and provide your valuable feedback. 

Regards,

### CODE REVIEW

File: delivery/media/hls/examples/live-program-replacement/main.js [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+import { LiveManifestTransformer} from './hls.js';
+
+const startDate = '2022-08-17T15:07:30.000Z';
+const endDate = '2022-08-17T15:07:50.000Z';
+
+const startDate1 = '2022-08-17T15:08:30.000Z';
+const endDate1 = '2022-08-17T15:08:50.000Z';
+
+// In this example the video policy is hardcoded.
+//You can also load the policy from EdgeKV or Property Manager and filterthe applicable policy based on the User Location Object.
```
Comment (bmatthew) [inline delivery/media/hls/examples/live-program-replacement/main.js:39]:
```suggestion
// You can also load the policy from EdgeKV or Property Manager and filter the applicable policy based on the User Location Object.
```

File: delivery/media/hls/examples/live-program-replacement/main.js [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+  #EXTINF:2.000,
+  http://example-ew-hls-mper.akamaized.net/Slate/Slate734s/audio_en/seg_0009.m4s
+  #EXTINF:2.000,
+  http://example-ew-hls-mper.akamaized.net/Slate/Slate734s/audio_en/seg_0010.m4s
+  #EXT-X-ENDLIST`
+}];
+
+const UNSAFE_RESPONSE_HEADERS = ['content-length', 'transfer-encoding', 'connection', 'vary',
+  'accept-encoding', 'content-encoding', 'keep-alive',
+  'proxy-authenticate', 'proxy-authorization', 'te', 'trailers', 'upgrade'];
```
Comment (bmatthew) [inline delivery/media/hls/examples/live-program-replacement/main.js:161]:
```suggestion
  'proxy-authenticate', 'proxy-authorization', 'te', 'trailers', 'upgrade', 'host'];
```

File: delivery/media/hls/examples/manifest-personalization/main.js [hunk 3]
```diff
@@ -0,0 +1,135 @@
+import { HLS } from './hls.js';
+import { httpRequest } from 'http-request';
+import { createResponse } from 'create-response';
+import URLSearchParams from "url-search-params";
+import { TextEncoderStream, TextDecoderStream } from 'text-encode-transform';
+import { ReadableStream, WritableStream } from 'streams';
+
+const UNSAFE_RESPONSE_HEADERS = ['content-length', 'transfer-encoding', 'connection', 'vary',
+  'accept-encoding', 'content-encoding', 'keep-alive',
+  'proxy-authenticate', 'proxy-authorization', 'te', 'trailers', 'upgrade'];
```
Comment (bmatthew) [inline delivery/media/hls/examples/manifest-personalization/main.js:10]:
```suggestion
  'proxy-authenticate', 'proxy-authorization', 'te', 'trailers', 'upgrade', 'host'];

```

File: delivery/media/hls/examples/bumper-insertion/main.js [hunk 4]
```diff
@@ -0,0 +1,115 @@
+import { HLS } from './hls.js';
+import { httpRequest } from 'http-request';
+import { createResponse } from 'create-response';
+import URLSearchParams from "url-search-params";
+import { TextEncoderStream, TextDecoderStream } from 'text-encode-transform';
+import { ReadableStream, WritableStream } from 'streams';
+
+const UNSAFE_RESPONSE_HEADERS = ['content-length', 'transfer-encoding', 'connection', 'vary',
+  'accept-encoding', 'content-encoding', 'keep-alive',
+  'proxy-authenticate', 'proxy-authorization', 'te', 'trailers', 'upgrade'];
```
Comment (bmatthew) [inline delivery/media/hls/examples/bumper-insertion/main.js:10]:
```suggestion
  'proxy-authenticate', 'proxy-authorization', 'te', 'trailers', 'upgrade', 'host'];

```

**Review by evan-hughes — APPROVED** [review]

File: delivery/media/hls/lib/hls.d.ts [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+    endDate: Date | string;
+    /**
+     * content to be replaced in manifest
+     */
+    content: string | HLS$1.types.MediaPlaylist;
+};
+
+/**
+ * LiveManifestTransformer class handles the replacement operation of blackslate segments in the hls manifest file for provided window.
+ * Assumtion:
```
[the remark this answers is not in the document]
Comment (sverekar) [reply delivery/media/hls/lib/hls.d.ts]: Fixed.

**Review by bmatthew — APPROVED** [review]

**Pull request merged**